src/aics-image-viewer/components/App/ChannelUpdater.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect } from "react";
+
+import { View3d, Volume } from "@aics/volume-viewer";
+
+import { controlPointsToLut } from "../../shared/utils/controlPointsToLut";
+import { ChannelState } from "../../shared/utils/viewerChannelSettings";
+import { UseImageEffectType } from "./types";
+
+interface ChannelUpdaterProps {
+  index: number;
+  channelState: ChannelState;
+  view3d: View3d;
+  image: Volume | null;
+  channelLoaded: boolean;
+}
+
+/**
+ * A component that doesn't render anything, but reacts to the provided `ChannelState`
+ * and keeps it in sync with the viewer.
+ */
+const ChannelUpdater: React.FC<ChannelUpdaterProps> = ({ index, channelState, view3d, image, channelLoaded }) => {
+  const { volumeEnabled, isosurfaceEnabled, isovalue, colorizeEnabled, colorizeAlpha, opacity, color, controlPoints } =
+    channelState;
+
+  // Effects to update channel settings should check if image is present and channel is loaded first
+  const useImageEffect: UseImageEffectType = (effect, deps) => {
+    useEffect(() => {
+      if (image && channelLoaded) {
```

Comment:
this `image` is the one that got passed in as a prop, right?  if so then I would either rename this one or the ones that are used in all the function args below.